run/service.go

```diff
@@ -82,6 +84,46 @@ func (s *Service) GetScheduledRun(ctx context.Context, namespace models.Namespac
 	return jobRun, err
 }
 
+func (s *Service) GetJobRunList(ctx context.Context, projectSpec models.ProjectSpec, jobSpec models.JobSpec, jobQuery *models.JobQuery) ([]models.JobRun, error) {
+	var jobRuns []models.JobRun
+	if jobQuery.OnlyLastRun {
+		return s.scheduler.GetJobRuns(ctx, projectSpec, jobQuery)
+	}
+	//validate job query
+	err := validateJobQuery(jobQuery, jobSpec)
+	if err != nil {
+		return jobRuns, err
+	}
+	//check for manual runs
+	jobInterval := jobSpec.Schedule.Interval
+	var expectedRuns []models.JobRun
+	if jobInterval != "" {
+		sch, err := cron.ParseCronSchedule(jobInterval)
+		if err != nil {
+			return jobRuns, fmt.Errorf("unable to parse the interval from DB %w", err)
+		}
+		//modify job query
+		jobQuery = modifyDateRange(jobQuery, jobSpec, sch)
```

Comment:
1. I believe there is an error here, while computing the expected runs, job runs to be created based on user input. not based on the scheduler specific times.
2. DateRange modification should be within scheduler and it shouldn't done in the runservice as it may not be applicable for all schedulers.
3. By default there is a limit of 100 runs that will be fetched through apis, i believe it is better to fetch all runs by firing multiple requests.
4. Fetching all manual runs is little more complicated  than i thought especially with airflow apis, let's postpone the support for this and throw an error to the user.

Reply:
1. expected runs are computed based on the user input only
2. DateRange modification is part of scheduler logic
3. if total entries are greater than the page limit, an error has been thrown
4. removed manual runs